### PR TITLE
7213/fix annual report generator

### LINKED ISSUE #7213: Some KEPs are not being picked by the KEP tool for annual reports
### Describe the issue

While working on [SIG Apps report](https://github.com/kubernetes/community/pull/7210) and [SIG CLI report](https://docs.google.com/document/d/1g3PCni-uVCSu4WdCaEbciQGky1J2Xm2GLSi-w56xq3A/edit?pli=1) I've noticed some of the KEPs not being picked by the tool. 

Specific examples for SIG-Apps are:
- 3140 - TimeZone support in CronJob
- 3017 - Pod Healthy Policy for PDB

for SIG-CLI:
- 3515 - Kubectl Explain OpenAPIv3
- 2227 - kubectl default container
- 1440 - Kubectl events

/cc @kubernetes/steering-committee @kubernetes/sig-contributor-experience

### PR DESCRIPTION
**Pull request by palnabarun**
**Which issue(s) this PR fixes**:

Fixes #7213 

- generator: fix KEP list generation for Annual Reports

/assign @Priyankasaggu11929 @MadhavJivrajani @cblecker

### CODE REVIEW

File: generator/annual-report/sig_report.tmpl [hunk 1]
```diff
@@ -16,13 +16,34 @@
...
 {{$releases := getReleases}}
 {{$owingsig := .Dir}}
+{{$stagedKeps := filterKEPs $owingsig $releases}}
+{{$alphaKeps := index $stagedKeps "alpha"}}
+{{$betaKeps := index $stagedKeps "beta"}}
+{{$stableKeps := index $stagedKeps "stable"}}
+<!--
+   Note: This list is generated from the KEP metadata in k/enhancements repository.
+      If you want find any discrepancy in the generated list here, please check the KEP metadata.
+      Please raise an issue in k/community if the KEP metadata is correct.
```
Comment (Priyankasaggu11929) [inline generator/annual-report/sig_report.tmpl:26]: Is the suggestion is to raise an issue in k/community "when the KEP metadata is matching the k/enhancements repo, but there is still a data discrepancy in the generated report"?

 If so, the current wording read a bit unclear.

**Comment by BenTheElder** [comment]
Generator change also LGTM

File: generator/annual-report/sig_report.tmpl (hunk 1 above)
Comment (Priyankasaggu11929) [reply generator/annual-report/sig_report.tmpl]: Yes. Thanks @palnabarun

**Comment by mrbobbytables** [comment]
We might have missed the opportunity to update the current reports with the KEPs =/
We def should get the generator update in no matter what

**Review by soltysh — COMMENTED** [review]
The generator changes lgtm, but at this point it time I'd not re-generate the 2022 reports.

**Comment by palnabarun** [comment]
> The generator changes lgtm, but at this point it time I'd not re-generate the 2022 reports.

Yes, that's the plan.

File: generator/annual-report/sig_report.tmpl [hunk 2]
```diff
@@ -16,13 +16,34 @@
 
 {{$releases := getReleases}}
 {{$owingsig := .Dir}}
+{{$stagedKeps := filterKEPs $owingsig $releases}}
+{{$alphaKeps := index $stagedKeps "alpha"}}
+{{$betaKeps := index $stagedKeps "beta"}}
+{{$stableKeps := index $stagedKeps "stable"}}
```
Comment (cblecker) [inline generator/annual-report/sig_report.tmpl:22]:
```suggestion
{{$releases := getReleases -}}
{{$owingsig := .Dir -}}
{{$stagedKeps := filterKEPs $owingsig $releases -}}
{{$alphaKeps := index $stagedKeps "alpha" -}}
{{$betaKeps := index $stagedKeps "beta" -}}
{{$stableKeps := index $stagedKeps "stable" -}}
```

File: generator/annual-report/sig_report.tmpl [hunk 3]
```diff
@@ -16,13 +16,34 @@
...
+<!--
+   Note: This list is generated from the KEP metadata in k/enhancements repository.
+      If you find any discrepancy in the generated list here, please check the KEP metadata.
+      Please raise an issue in k/community, if the KEP metadata is correct but the generated list is incorrect.
+-->
```
Comment (cblecker) [inline generator/annual-report/sig_report.tmpl:27]:
```suggestion
-->

```

**Review by cblecker — COMMENTED** [review]
Some whitespace fixes, but otherwise lgtm

**Review by soltysh — APPROVED** [review]
/lgtm

**Comment by palnabarun** [comment]
Thanks @cblecker for the review!

**Comment by palnabarun** [comment]
@soltysh @cblecker require another lgtm now.

File: generator/annual-report/sig_report.tmpl [hunk 4]
```diff
@@ -14,15 +14,37 @@
    -
    -
 
-{{$releases := getReleases}}
-{{$owingsig := .Dir}}
+{{$releases := getReleases -}}
+{{$owingsig := .Dir -}}
```
Comment (justaugustus) [inline generator/annual-report/sig_report.tmpl:18]:
```suggestion
{{$owningSIG := .Dir -}}
```

?
Same throughout.

File: generator/annual-report/sig_report.tmpl [hunk 5]
```diff
@@ -14,15 +14,37 @@
...
+{{$stagedKeps := filterKEPs $owingsig $releases -}}
+{{$alphaKeps := index $stagedKeps "alpha" -}}
+{{$betaKeps := index $stagedKeps "beta" -}}
+{{$stableKeps := index $stagedKeps "stable" -}}
+<!--
+   Note: This list is generated from the KEP metadata in k/enhancements repository.
```
Comment (justaugustus) [inline generator/annual-report/sig_report.tmpl:24]:
```suggestion
   Note: This list is generated from the KEP metadata in kubernetes/enhancements repository.
```

File: generator/annual-report/sig_report.tmpl [hunk 6]
```diff
@@ -14,15 +14,37 @@
...
+      If you find any discrepancy in the generated list here, please check the KEP metadata.
+      Please raise an issue in k/community, if the KEP metadata is correct but the generated list is incorrect.
```
Comment (justaugustus) [inline generator/annual-report/sig_report.tmpl:26]:
```suggestion
      Please raise an issue in kubernetes/community, if the KEP metadata is correct but the generated list is incorrect.
```

**Review by justaugustus — COMMENTED** [review]
Non-blocking: minor style nits

**Comment by BenTheElder** [comment]
+1 to outstanding comments

**Comment by palnabarun** [comment]
Resolved all comments.

**Pull request merged**